bats/sql-batch.bats

```diff
@@ -120,3 +120,26 @@ SQL
     [ "$status" -eq 0 ]
     [[ "$output" =~ "working tree clean" ]] || false
 }
+
+@test "batch mode detects subqueries and decides not to do batch insert." {
+  # create the second table.
+  dolt sql << SQL
+CREATE TABLE test2 (
+  pk bigint NOT NULL,
+  c1 bigint,
+  PRIMARY KEY (pk)
+);
+SQL
+
+  run dolt status
+  [ "$status" -eq 0 ]
+
+  # Create the table and base subquery on recently inserted row.
+  run dolt sql << SQL
+INSERT INTO TEST VALUES (1,1,0,0,0,0);
```

Comment:
You should have a straight tuple value insert into test2 before you do a subselect insert.